src/share/import.js

```diff
@@ -0,0 +1,77 @@
+import fs from '../lib/fs';
+import Buffer from '../lib/buffer';
+import Path from '../lib/path';
+import WebTorrent from 'webtorrent';
+const sh = new fs.Shell();
+
+document.getElementById('torrentId').value = window.location.search.substring(
+    1
+);
+document.getElementById('btnDownload').addEventListener('click', startDownload);
+
+const downloadAllTorrentFiles = torrent =>
+    Promise.all(
+        torrent.files.map(
+            file =>
+                new Promise((resolve, reject) => {
+                    console.log('file: ', file);
+                    var filename = Path.join('/', file.path);
+                    file.getBuffer((err, buffer) => {
+                        if (err) {
+                            writeError();
+                            return reject(err);
+                        }
+                        sh.mkdirp(Path.dirname(filename), err => {
+                            if (err) {
+                                console.error(

```

Comment:
Let's add this `console.error` logic to `writeError` so you can shorten this code and not repeat yourself.  DRY this out.